pkg/internal/server/grpc/get_metadata_allocated.go

```diff
@@ -48,12 +50,26 @@ func (s *Server) GetMetadataAllocated(req *api.GetMetadataAllocatedRequest, stre
 	}
 
 	// Invoke the CSI Driver's GetMetadataDelta gRPC and stream the response back to client
+	klog.FromContext(ctx).V(HandlerTraceLogLevel).Info("calling CSI driver", "snapshotId", csiReq.SnapshotId)
 	csiStream, err := csi.NewSnapshotMetadataClient(s.csiConnection()).GetMetadataAllocated(ctx, csiReq)
 	if err != nil {
 		return err
 	}
 
-	return s.streamGetMetadataAllocatedResponse(stream, csiStream)
+	return s.streamGetMetadataAllocatedResponse(ctx, stream, csiStream)
+}
+
+// getMetadataAllocatedContextWithLogger returns the stream context with an embedded
+// contextual logger primed with a description of the request.
+func (s *Server) getMetadataAllocatedContextWithLogger(req *api.GetMetadataAllocatedRequest, stream api.SnapshotMetadata_GetMetadataAllocatedServer) context.Context {
+	return klog.NewContext(stream.Context(),
+		klog.LoggerWithValues(klog.Background(),
+			"op", s.OperationID("GetMetadataAllocated"),
+			"namespace", req.Namespace,
+			"snapshotName", req.SnapshotName,
+			"startingOffset", req.StartingOffset,
+			"maxResults", req.MaxResults,
```

Comment:
I was wondering whether we only need to put the operation ID in the context and have a separate call to log the request parameters but then I discarded that notion for the following reasons:
- logging is typically for errors and not tracing
- log files can roll over during long operations and the request parameters can get lost

Reply:
Is it going to print the parameters each time we log with this context?

Reply:
Yes.  See the test output above. It looks like this:
> E0918 15:27:03.434327  837547 auth.go:41] "unauthenticated user" op="GetMetadataDelta-2" namespace="namespace" baseSnapshotName="snap-1" targetSnapshotName="snap-2" startingOffset=0 maxResults=0 userInfo="nil"

Reply:
`unauthentic user` seems more generic. If we are able to mention specific reason, that would be better.